leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java

```diff
@@ -62,22 +71,37 @@ private SecurityInfo(String endpoint, String identity, byte[] preSharedKey, Publ
     public static SecurityInfo newPreSharedKeyInfo(String endpoint, String identity, byte[] preSharedKey) {
         Validate.notEmpty(identity);
         Validate.notNull(preSharedKey);
-        return new SecurityInfo(endpoint, identity, preSharedKey, null, false);
+        return new SecurityInfo(endpoint, identity, preSharedKey, null, false, null);
     }
 
     /**
      * Construct a {@link SecurityInfo} when using DTLS with Raw Public Key (RPK).
      */
     public static SecurityInfo newRawPublicKeyInfo(String endpoint, PublicKey rawPublicKey) {
         Validate.notNull(rawPublicKey);
-        return new SecurityInfo(endpoint, null, null, rawPublicKey, false);
+        return new SecurityInfo(endpoint, null, null, rawPublicKey, false, null);
     }
 
     /**
      * Construct a {@link SecurityInfo} when using DTLS with an X509 Certificate.
      */
     public static SecurityInfo newX509CertInfo(String endpoint) {
-        return new SecurityInfo(endpoint, null, null, null, true);
+        return new SecurityInfo(endpoint, null, null, null, true, null);
+    }
+
+    /**
+     * Construct a {@link SecurityInfo} when using OSCORE.
+     */
+    public static SecurityInfo newOSCoreInfo(String endpoint, String identity, OSCoreCtx oscoreCtx) {
+        Validate.notEmpty(identity);
+        Validate.notNull(identity);
+        Validate.notNull(oscoreCtx);
+
+        // Add the OSCORE Context to the context database
+        HashMapCtxDB db = HashMapCtxDB.getInstance();
+        db.addContext(oscoreCtx);
```

Comment:
Here this is a problem for me.
Create a new securityInfo should not modify the state of the application.